Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/BrowserLog/BrowserLogUtility.cs

```diff
@@ -0,0 +1,152 @@
+    return minimumLogLevel ?? WebTestSettings.Current.CheckBrowserLogLevel;
+  }
+
+  /// <summary>
+  /// Gets the browser log entries that are not filtered out by a <see cref="IgnoreBrowserLogMessageAttribute"/>.
+  /// </summary>
+  public static IReadOnlyCollection<Func<BrowserLogEntry, bool>> GetBrowserLogIgnoredEntriesPredicates (Test test)
+  {
+    ArgumentUtility.CheckNotNull(nameof(test), test);
+
+    var delegates = new List<Func<BrowserLogEntry, bool>>();
+    ITest currentTest = test;
+    while (currentTest != null)
+    {
+      delegates.AddRange(currentTest.Properties[IgnoreBrowserLogMessageAttribute.PropertyKey].Cast<Func<BrowserLogEntry, bool>>());
+      currentTest = currentTest.Parent;
+    }
+
+    return delegates;
+  }
+
+  /// <summary>
+  /// Gets the browser log entries that indicate that the test should fail, because their <see cref="LogLevel"/> is at least <paramref name="minLogLevel"/>, and they do not match
+  /// <paramref name="ignoreLogEntry"/>.
+  /// </summary>
+  public static IReadOnlyCollection<BrowserLogEntry> GetUnexpectedBrowserLogEntries (
+      [NotNull] IBrowserSession session,
+      LogLevel minLogLevel,
+      [NotNull] Func<BrowserLogEntry, bool> ignoreLogEntry)
+  {
+    ArgumentUtility.CheckNotNull(nameof(session), session);
+    ArgumentUtility.CheckNotNull(nameof(ignoreLogEntry), ignoreLogEntry);
+
+    var remainingRetries = 50;
+    IReadOnlyCollection<BrowserLogEntry> logEntries;
+    do
+    {
+      Thread.Sleep(50);
+      logEntries = session.GetBrowserLogs();
+      remainingRetries--;
+    } while (remainingRetries > 0 && logEntries.Count == 0);
+
+    logEntries = logEntries
+        .Where(e => e.Level >= minLogLevel)
+        .Where(entry => !ignoreLogEntry(entry))
+        .ToList();
+
+    return logEntries;
+  }
+
+  /// <summary>
+  /// If browser log checking is active, this method checks whether there are unexpected browser log entries that meet the minimum <see cref="LogLevel"/>. If so, the test is set
+  /// to <see cref="ResultState.Failure"/>, with a message containing the relevant browser log entries.
+  /// </summary>
+  /// <remarks>
+  /// Call this in the <see cref="TearDownAttribute"/> method.
+  /// </remarks>
+  public static void CheckBrowserLog ([NotNull] IBrowserSession session, [NotNull] TestExecutionContext context)
+  {
+    ArgumentUtility.CheckNotNull(nameof(session), session);
+    ArgumentUtility.CheckNotNull(nameof(context), context);
+
+    if (context.CurrentResult.ResultState != ResultState.Success)
+      return;
+
+    var isActive = IsBrowserLogCheckActive(context.CurrentTest);
+    if (!isActive)
+      return;
+
+    var minLogLevel = GetMinimumLogLevel(context.CurrentTest);
+    var ignoreLogEntries = GetBrowserLogIgnoredEntriesPredicates(context.CurrentTest);
+    bool IgnoreLogEntry (BrowserLogEntry entry) => ignoreLogEntries.Any(func => func(entry));
```

Comment:
Local functions should not be in the middle of other code. usally, we put them at the end of the method. Otherwise, there's no reason to not simply make it an anonymous function.

Given that there's a closure, I would recommend we use an anonymous function.